<launch>
  <!-- subscribes to cmd_vel topic -->
  <!-- publishes odom topic with base_link frame -->
  <!-- broadcasts odom->base_link transform -->
  <arg name="param" default="value" />
  <arg name="odom_frame" default="/caato0_tf/odom" />
  <arg name="base_frame" default="/caato_0/base_link" />
  <node pkg="plc_modbus_controllers" type="driver_modbus_node" name="driver_modbus_node" output="screen">
    <!-- enable broadcast of odom tf -->
    <param name="pub_odom_tf" value="false" />
    <!-- specify odom frame -->
    <param name="odom_frame" value="$(arg odom_frame)" />
    <param name="odom_topic" value="/caato_0/odom" />
    <!-- specify base frame -->
    <param name="base_frame" value="$(arg base_frame)" />
    <!-- specify cmd_vel topic -->
    <param name="cmdvel_topic" value="/caato_0/cmd_vel" />
    <!-- specify regs_write topic -->
    <param name="regs_write_topic" value="/caato_0/modbus/regs_write" />
    <!-- specify roboteq sensors topic -->
    <param name="sensors_topic" value="/modbus/roboteq_sensors" />
    <!-- specify whether to use open-loop motor speed control (as opposed to closed-loop)-->
    <param name="open_loop" value="false" />
    <!-- specify robot wheel circumference in meters  -->
    <param name="wheel_circumference" value="0.491" />
    <!-- specify robot track width in meters  -->
    <param name="track_width" value="0.47" />
    <!-- specify pulse-per-revolution for motor encoders  -->
    <param name="encoder_ppr" value="10240" />
    <!-- specify counts-per-revolution for motor encoders (ppr*4 for quadrature encoders)  -->
    <param name="encoder_cpr" value="40960" />
    <!-- specify maximum motor current (amps) -->
    <param name="max_amps" value="5.0" />
    <!-- specify maximum motor speed (rpm) -->
    <param name="max_rpm" value="100" />
  </node>
</launch>
